tls/s2n_crypto.c

```diff
@@ -119,3 +119,19 @@ S2N_RESULT s2n_crypto_parameters_switch(struct s2n_connection *conn)
 
     return S2N_RESULT_OK;
 }
+
+int s2n_connection_get_master_secret(const struct s2n_connection *conn,
+        uint8_t *secret_bytes, size_t max_size)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(secret_bytes);
+    POSIX_ENSURE(max_size >= S2N_TLS_SECRET_LEN, S2N_ERR_INSUFFICIENT_MEM_SIZE);
+    POSIX_ENSURE(conn->actual_protocol_version < S2N_TLS13, S2N_ERR_INVALID_STATE);
```

Comment:
In the documentation we mention that this is a TLS 1.2 specific feature. Should we switch this to `actual_protocol_version == S2N_TLS12`? I'm assuming that bad things happen if this is called with TLS 1.1 or friends.

Reply:
No, you can also call it for earlier versions and not have anything worse happen than if you call it for TLS1.2. The big distinction is between pre-TLS1.3 and TLS1.3, when the key schedule changed and TLS actually started caring about keeping secret material separate.

I updated the documentation to not call out TLS1.2 specifically, and expanded my testing to include earlier versions.